glslc/test/option_masm.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2016 The Shaderc Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import expect
+from glslc_test_framework import inside_glslc_testsuite
+from placeholder import FileShader
+
+MINIMAL_SHADER = '#version 140\nvoid main() {}'
+
+
+@inside_glslc_testsuite('OptionMasm')
+class TestSingleFileWorksWithDashC(expect.ValidFileContents):
+    """Tests that -masm=c-init-list works with -c for single input file. SPIRV
+    binary code output should be emitted as a C-style initializer list in the
+    output file.
+    """
+    shader = FileShader(MINIMAL_SHADER, '.vert')
+    glslc_args = [shader, '-c', '-masm=c-init-list', '-o', 'output_file']
+    target_filename = 'output_file'
+    expected_file_contents = [
+        '{0x07230203,0x00010000,0x00080001,0x00000006,0x00000000,0x00020011,'
```

Comment:
This is going to be a very fragile test as Glslang evolves.
You should only check the up through the first word since the magic word won't change.